lib/src/git_backend.rs

```diff
@@ -859,6 +867,8 @@ fn import_extra_metadata_entries_from_heads(
     head_ids: &HashSet<&CommitId>,
     uses_tree_conflict_format: bool,
 ) -> BackendResult<()> {
+    let shallow_commits = git_repo.shallow_commits().ok().flatten();
```

Comment:
nit: better to map error to `BackendError::Other`, but that can be addressed later. We'll get another read error anyway.